content/src/content/docs/learn/tutorials/calling-an-api.mdx

```diff
@@ -0,0 +1,452 @@
+"context" object. You can think of this as a map of services, with the key being
+a unique identifier for the service, and the value being the service itself.
+
+Effect uses a type called a `Tag` as the unique identifier for services. The
+way that tags are defined is quite complex, but this complexity is necessary
+in order to get the level of type safety that Effect provides.
+
+```ts twoslash showLineNumbers=false
+import { Context, Effect } from "effect"
+//---cut---
+class FavouriteNumber extends Context.Tag("FavouriteNumber")<
+  FavouriteNumber,
+  number
+>() {}
+```
+
+You will typically make tags once per service and then forget about them, so
+don't worry too much about memorising this syntax. The important things to note
+are that:
+
+- The service is called `FavouriteNumber`.
+- It returns a `number`.
+
+To use this service in an `Effect` we can use `Effect.gen` as we saw in the
+previous section.
+
+```ts twoslash showLineNumbers=false
+import { Context, Effect } from "effect"
+class FavouriteNumber extends Context.Tag("FavouriteNumber")<
+  FavouriteNumber,
+  number
+>() {}
+//---cut---
+const printNum = Effect.gen(function* () {
+  const n = yield* FavouriteNumber
+  console.log(`My favourite number is ${n}`)
+})
+```
+
+Lastly, we won't be able to run this `Effect` until we provide an implementation
+of the `FavouriteNumber` service to it, which we can do with
+`Effect.provideService`.
+
+```ts twoslash showLineNumbers=false
+import { Context, Effect } from "effect"
+class FavouriteNumber extends Context.Tag("FavouriteNumber")<
+  FavouriteNumber,
+  number
+>() {}
+
+const printNum = Effect.gen(function* () {
+  const n = yield* FavouriteNumber
+  console.log(`My favourite number is ${n}`)
+})
+//---cut---
+Effect.runPromise(
+  printNum.pipe(Effect.provideService(FavouriteNumber, 42))
+)
+// => My favourite number is 42
+
+Effect.runPromise(
+  printNum.pipe(Effect.provideService(FavouriteNumber, 100))
+)
+// => My favourite number is 100
+```
+
+Providing a service to an Effect ends up changing its type, too.
+
+```ts twoslash showLineNumbers=false
+import { Context, Effect } from "effect"
+class FavouriteNumber extends Context.Tag("FavouriteNumber")<
+  FavouriteNumber,
+  number
+>() {}
+//---cut---
+const needsRequirements = Effect.gen(function* () {
+  //  needsRequirements: Effect<void, never, FavouriteNumber>
+  const n = yield* FavouriteNumber
+  console.log(`My favourite number is ${n}`)
+})
+
+const hasRequirements = needsRequirements.pipe(
+  //  hasRequirements: Effect<void, never, never>
+  Effect.provideService(FavouriteNumber, 42)
+)
+```
+
+We go from:
+
+``` showLineNumbers=false
+Effect<void, never, FavouriteNumber>
+```
+
+To:
+
+``` showLineNumbers=false
+Effect<void, never, never>
+```
+
+So each Effect can be thought of as a self-contained program. Its requirements
+are its inputs, its success value is its output, and it can error. Effect forces
```

Comment:
Technically speaking, the output of an Effect program can be either its success value or its possible errors.